### PR TITLE
fix(component-testing): Show a hint if no specs were found in the list

### PR DESCRIPTION
**Pull request by dmtrKovalenko**
When the project is empty and there are no specs cypress is saying "please select specs from specs list", but no specs is displayed. Here we fix this. 

Check UI change in Percy snapshot  -> https://percy.io/cypress-io/cypress/builds/9596545/changed/541617631?browser=chrome&browser_ids=16&subcategories=unreviewed%2Cchanges_requested&viewLayout=side-by-side&viewMode=new&width=1280&widths=400%2C1280

ui

### CODE REVIEW

File: packages/runner-ct/src/app/SpecList/SpecList.tsx [hunk 1]
```diff
@@ -194,6 +195,10 @@ export const FileTree: React.FC<FileTreeProps> = (props) => {
     return render(match.indexes)
   }
 
+  if (props.specs.length < 1) {
+    return <NoSpec message="No spec found" />
```
Comment (agg23) [inline packages/runner-ct/src/app/SpecList/SpecList.tsx:199]: super nit: Maybe centralize the strings in `NoSpec`, and choose one through props.

**Review by agg23 — DISMISSED** [review]

File: packages/runner-ct/src/app/RunnerCt.tsx [hunk 2]
```diff
@@ -199,16 +200,37 @@ const RunnerCt = namedObserver('RunnerCt',
...
+            state.specs.length < 1 ? (
+              <NoSpec message="No specs found">
+                <p className={styles.noSpecsDescription}>
+                  Create a new spec file in
+                  {' '}
+                  <span className={styles.folder}>
+                    {
+                      props.config.componentFolder
+                        ? props.config.componentFolder.replace(props.config.projectRoot, '')
+                        : 'component specs folder'
```
[the remark this answers is not in the document]
Comment (dmtrKovalenko) [reply packages/runner-ct/src/app/RunnerCt.tsx:213]: Yeah please fix my bad English :) 

```suggestion
                        : 'the component specs folder'
```

Is this OK?

[the remark this answers is not in the document]
Comment (agg23) [reply packages/runner-ct/src/app/RunnerCt.tsx:213]: Yes, looks good.

File: packages/runner-ct/src/app/RunnerCt.tsx [hunk 3]
```diff
@@ -199,16 +200,37 @@ const RunnerCt = namedObserver('RunnerCt',
...
-            specs={props.state.specs}
-            selectedFile={state.spec ? state.spec.relative : undefined}
-            focusSpecList={focusSpecsList}
-            searchRef={searchRef}
-            className={cs(styles.specsList, {
-              'display-none': hideSpecsListIfNecessary(state),
-            })}
-            onFileClick={runSpec}
-          />
+          {
```
Comment (lmiller1990) [inline packages/runner-ct/src/app/RunnerCt.tsx:203]: I'm interested in the team's opinion on this style, especially @dmtrKovalenko and @agg23 who have worked on very big React code-bases. In my past experience we tried to keep the `return` function simple and logic free, instead opting for a variable. Eg:

```tsx
const specListContent = state.specs.length < 1 
  ? <NoSpec />
  : <SpecList />

// ...

return (
  <div>
    {specListContent}
  </div>
)
```

I don't have a super strong opinion, while I do personally prefer this style it's not really a big deal either way. What do you guys think?

**Review by lmiller1990 — APPROVED** [review]
I left a comment, more of a question.

The code itself seems fine and I tested it locally too, so ✅  for me.

image

File: packages/runner-ct/src/app/RunnerCt.tsx (hunk 3 above)
Comment (dmtrKovalenko) [reply packages/runner-ct/src/app/RunnerCt.tsx:203]: I will prefer inline option because it takes some time to understand what exactly rendered inside div, especially if jsx is pretty big and you need to go to definition to view specListContent.

Comment (dmtrKovalenko) [reply packages/runner-ct/src/app/RunnerCt.tsx:203]: We can reduce amount of inlined jsx by making an element before return. 

in order to make

```js
state.specs.length < 1 
  ? createNewSpecFiller
  : <SpecsList /> 
```

Comment (agg23) [reply packages/runner-ct/src/app/RunnerCt.tsx:203]: Agreed with Dmitriy. JSX can be quite hard to parse without inline flow control, and in my opinion the flow control is neat enough that it's easily understood.

**Comment by lmiller1990** [comment]
Looks like this already got merged, it's in develop

**Comment by brian-mann** [comment]
This is actually not a `chore` - it's a `fix` because it's a user facing change that belongs in the changelog. `chore` is anything that is **not** user facing and does not belong on the changelog
